File: backend/models/ServiceRequest.js
const mongoose = require("mongoose");

const serviceRequestSchema = new mongoose.Schema({
    tenantName: { type: String, required: true },
    unit: { type: String, required: true },
    type: { type: String, required: true, enum: ["Parking", "Rooftop"] },
    details: { type: String },
    date: { type: Date, default: Date.now },
    schedule: {
        date: Date,
        time: String,
    },
    vehicle: { type: String },
    plateNumber: { type: String },
    status: { type: String, default: "Pending", enum: ["Pending", "Approved", "Denied"] },
    denialReason: { type: String, default: "" } // NEW: store denial description
});

module.exports = mongoose.model("ServiceRequest", serviceRequestSchema);

File: app/src/main/java/com/regalrooms/tenant/models/ServiceRequest.java
package com.regalrooms.tenant.models;

import com.google.gson.annotations.SerializedName;
import java.util.Date;

public class ServiceRequest {
    @SerializedName("_id")
    private String id;
    private String tenantName;
    private String unit;
    private String type;
    private String vehicle;
    private String plateNumber;
    private Schedule schedule;
    private String details;
    private String status;
    private String denialReason;

    public static class Schedule {
        private String date;
        private String time;

        public String getDate() { return date; }
        public String getTime() { return time; }
    }

    // Getters and Setters
    public String getId() { return id; }
    public String getTenantName() { return tenantName; }
    public String getUnit() { return unit; }
    public String getType() { return type; }
    public String getVehicle() { return vehicle; }
    public String getPlateNumber() { return plateNumber; }
    public Schedule getSchedule() { return schedule; }
    public String getDetails() { return details; }
    public String getStatus() { return status; }
    public String getDenialReason() { return denialReason; }
}